rotkehlchen/api/v1/schemas.py

```diff
@@ -2115,6 +2116,25 @@ class AssetsReplaceSchema(Schema):
     source_identifier = fields.String(required=True)
     target_asset = AssetField(required=True, expected_type=Asset, form_with_incomplete_data=True)
 
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        try:
+            source_is_evm_token = Asset(data['source_identifier']).is_evm_token()
+        except UnknownAsset:
+            can_merge = True  # we allow to merge assets that are not in the globaldb
+        else:
+            can_merge = not (source_is_evm_token and data['target_asset'].is_evm_token())
+
+        if can_merge is False:
```

Comment:
This logic also can improve.

We need the source to not be an evm token.
The target can be an evm token I think. Say an asset got merged into an evm token so for ease of use the user wants to replace all old appearances of the non evm token with the evm token.

And then since the source token can also not be in the global DB we need to check for the evm chain directive instead to detect the problem also if it's not in the DB.

```suggestion
        if data['source_identifier'].startswith(EVM_CHAIN_DIRECTIVE)
```